### PR TITLE
storage: calculate last_change_ts in prewrite

### PR DESCRIPTION
**Pull request by sticnarf**
### What is changed and how it works?

Issue Number: Ref #13694

What's Changed:

```commit-message
This commit implements the different cases when last_change_ts
is calculated in prewrite:
1. Inherit from the pessimistic lock
2. Calculate it when checking the new version
3. Amend the pessimistic lock
```

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: src/storage/txn/actions/prewrite.rs [hunk 1]
```diff
@@ -727,6 +746,8 @@ fn amend_pessimistic_lock<S: Snapshot>(
             }
             .into());
         }
+        (mutation.last_change_ts, mutation.versions_to_last_change) =
+            write.next_last_change_info(*commit_ts);
```
Comment (MyonKeminta) [inline src/storage/txn/actions/prewrite.rs:750]: Do we need to check the `write`'s type here?

**Comment by sticnarf** [comment]
...I give up comparing `last_change_ts` in `test_mess`. It's tedious and not so meaningful...

**Review by MyonKeminta — APPROVED** [review]

**Comment by sticnarf** [comment]
PTAL @ekexium

File: src/storage/txn/actions/prewrite.rs [hunk 2]
```diff
@@ -440,6 +454,11 @@ impl<'a> PrewriteMutation<'a> {
             self.txn_props.txn_size,
             self.min_commit_ts,
         );
+        // Only Lock needs to record `last_change_ts` in its write record, write records
+        // of other types themselves are effective changes.
```
Comment (ekexium) [inline src/storage/txn/actions/prewrite.rs:458]: Nit: ROLLBACK doesn't, though it doesn't matter at all

**Review by ekexium — APPROVED** [review]

File: src/storage/txn/actions/prewrite.rs (hunk 2 above)
Comment (sticnarf) [reply src/storage/txn/actions/prewrite.rs:458]: Ah, I mean the possible LockType here. We don't have a `ROLLBACK` lock.

**Comment by TonsnakeLin** [comment]
I find it updates the `(last_change_ts, versions_to_last_change)` in the function `check_for_newer_version` which is  not called in so scenario such as that the kind of transaction is  TransactionKind::Optimistic(true). It will make some locks have   `(last_change_ts, versions_to_last_change)` and some locks have not on the same key. Did we notice this?

**Comment by sticnarf** [comment]
> I find it updates the `(last_change_ts, versions_to_last_change)` in the function `check_for_newer_version` which is not called in so scenario such as that the kind of transaction is TransactionKind::Optimistic(true). It will make some locks have `(last_change_ts, versions_to_last_change)` and some locks have not on the same key. Did we notice this?

`skip_constraint_check` is legacy. The client never sets this configuration (and will not set it in the foreseeable future).

**Pull request merged**